benchmarks/gravity_prem/prem.cc

```diff
@@ -0,0 +1,176 @@
+              else if ((position_point_sph[0] > 5701.e3) && (position_point_sph[0] <= 5771.e3))
+                out.densities[i]=(5.3197-1.4836*x)*1e3;
+              else if ((position_point_sph[0] > 5771.e3) && (position_point_sph[0] <= 5971.e3))
+                out.densities[i]=(11.2494-8.0298*x)*1e3;
+              else if ((position_point_sph[0] > 5971.e3) && (position_point_sph[0] <= 6151.e3))
+                out.densities[i]=(7.1089-3.8045*x)*1e3;
+              else if ((position_point_sph[0] > 6151.e3) && (position_point_sph[0] <= 6346.e3))
+                out.densities[i]=(2.6910+0.6924*x)*1e3;
+              else if ((position_point_sph[0] > 6346.e3) && (position_point_sph[0] <= 6356.e3))
+                out.densities[i]=2.900*1e3;
+              else if ((position_point_sph[0] > 6356.e3) && (position_point_sph[0] <= 6368.e3))
+                out.densities[i]=2.600*1e3;
+              else
+                out.densities[i]=1.020*1e3;
+
+              out.compressibilities[i] = 0;
+              out.specific_heat[i] = 0;
+              out.thermal_expansion_coefficients[i] = 0;
+              out.thermal_conductivities[i] = 0.0;
+              out.viscosities[i] = 0;
+            }
+        }
+        /**
+         * @}
+         */
+
+        /**
+         * @name Qualitative properties one can ask a material model
+         * @{
+         */
+
+        /**
+         * Return whether the model is compressible or not.
+         * Incompressibility does not necessarily imply that the density is
+         * constant; rather, it may still depend on temperature or pressure.
+         * In the current context, compressibility means whether we should
+         * solve the continuity equation as $\nabla \cdot (\rho \mathbf u)=0$
+         * (compressible Stokes) or as $\nabla \cdot \mathbf{u}=0$
+         * (incompressible Stokes).
+         */
+        virtual bool is_compressible () const;
+        /**
+         * @}
+         */
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &)
+        {
+          // Declare dependencies on solution variables
+          this->model_dependence.viscosity = MaterialModel::NonlinearDependence::none;
+          this->model_dependence.density = MaterialModel::NonlinearDependence::none;
+          this->model_dependence.compressibility = MaterialModel::NonlinearDependence::none;
+          this->model_dependence.specific_heat = MaterialModel::NonlinearDependence::none;
+          this->model_dependence.thermal_conductivity = MaterialModel::NonlinearDependence::none;
+        }
+
+        /**
+         * @name Reference quantities
+         * @{
+         */
+        virtual double reference_viscosity () const;
+        /**
+         * @}
+         */
+
+    };
+
+
+    template <int dim>
+    double
+    PremDensity<dim>::
+    reference_viscosity () const
```

Comment:
Since you already have the implementation of the `evaluate` function in the class declaration above, you may as well move this function and the `is_compressible` function in there as well to be consistent. Saves some lines.

Reply:
@gassmoeller I am not sure where exactly I have to move those blocks. And if I move them, do I need to change the format of the blocks? I'll need help on this one.

Reply:
You can take a look at how the `evaluate` function is implemented. There is only one definition of that function (not two like for `reference_viscosity` in line 135 and 146) and the one definition is further up inside the `class PremDensity{ };` block. If you move this function to line 135, and make the function look like the `reference_viscosity` function there (remove the class name and the template line), then it should work just the same as now. The same for the `is_compressible` function in line 155 and 111 (move it from 155 to 111). Does that help?

Reply:
@gassmoeller thanks! I have moved those lines.